src/systemd/sd-event.h

```diff
@@ -69,7 +69,7 @@ typedef int (*sd_event_handler_t)(sd_event_source *s, void *userdata);
 typedef int (*sd_event_io_handler_t)(sd_event_source *s, int fd, uint32_t revents, void *userdata);
 typedef int (*sd_event_time_handler_t)(sd_event_source *s, uint64_t usec, void *userdata);
 typedef int (*sd_event_signal_handler_t)(sd_event_source *s, const struct signalfd_siginfo *si, void *userdata);
-#if defined __USE_POSIX199309 || defined __USE_XOPEN_EXTENDED
+#if defined _GNU_SOURCE || defined _POSIX_C_SOURCE && _POSIX_C_SOURCE >= 199309L
```

Comment:
Why `_GNU_SOURCE`? Isn't that redundant with `_POSIX_C_SOURCE >= 199309L`?

Please drop this part: `defined _POSIX_C_SOURCE &&`. If an identifier is not defined, it is replaced by 0 in comparisons.

See b2542bf9ab6cce97fcb67ff9536e1062c70b5b11 for the motivation for the way this check was made. Indeed, it's using a detail of the glibc implementation, and the check you are proposing (at least the second part), follows the standard more closely.